code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1116,9 +1116,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	glass_icon_state = "hearty_punch"
 	glass_name = "Hearty Punch"
 	glass_desc = "Aromatic beverage served piping hot. According to folk tales it can almost wake the dead."
+	var/active = FALSE
 
 /datum/reagent/consumable/ethanol/hearty_punch/on_mob_life(mob/living/M)
-	if(M.health <= 0)
+	if(!active & M.health <= 0)
+		active = TRUE
+		M.visible_message("<span class='warning'>[M.name] collapses, but [M.p_they()] look like they're determined to get back up.</span>",
+			"<span class='warning'>You fall over, but hear a soothing, gruff voice urging you back on your feet.</span>")
+	if(active & M.health >= 50)
+		active = FALSE
+		M.visible_message("<span class='warning'>[M.name] starts to get back up and dust [M.p_them()]self off.</span>",
+			"<span class='notice'>You start to get back up and brush yourself off, feeling better.</span>")
+		M.reagents.remove_reagent("hearty_punch",M.reagents.get_reagent_amount("hearty_punch"))
```

Comment:
ummm either `M.reagents.del_reagent(id)` or `M.reagents.remove_reagent(id,volume)` up to personal preference

Reply:
Wasn't aware `del_reagent` was a thing! Will do.